tests/test_window.py

```diff
@@ -290,3 +290,11 @@ def test_select_layout_accepts_no_arg(server: Server, session: Session) -> None:
 
     window = session.new_window(window_name="test_window")
     window.select_layout()
+
+
+def test_empty_window_name(session: Session) -> None:
+    session.set_option("automatic-rename", "off")
+    window = session.new_window(window_name="''", attach=True)
+
+    assert window == session.attached_window
+    assert window.get("window_name") == "''"
```

Comment:
What's interesting is this test always passes.

If you remove from the changes from session.py and run:

```console
poetry run py.test tests/test_window.py::test_empty_window_name -vv -x -s
```

Do they still pass?

The implication is maybe we're looking at the wrong spot 🤔

Reply:
How do you suggest to test this?

Reply:
I will need to look at this during the weekend when I have more time, or tomorrow evening if I have time 

(i work during the day US time)

Reply:
Yeah no rush

Reply:
If you haven't yet you can clone libvcs and tmuxp, then do `pip install --user -e ~/path/to/tmuxp`, `pip install --user -e ~/path/to/libtmux` and you should be able to test it by hand.

If you have `entr(1)` installed:

```console
make watch_test test='tests/test_window.py::test_empty_window_name -vv -x -s'
```

If no `entr(1)`:

```console
env PYTEST_ADDOPTS="-s -x -vvv tests/test_window.py::test_empty_window_name" poetry run make start
```

I actually don't know why the test passes always yet

Reply:
I read the code and it looks like `Window.new_window()` ran a command against the OS and then forgets about it. What I want to know is where does `Window.get()` gets the window_name from. I can't find `Window.get()` in the code. Do you mind pointing me to it? I can't find it in the inherited class neither.

I order to test this we need to be able to ask tmux itself for the window name and compare it against the expected value. Is there a way to ask tmux server for the current window name vs getting it from `Window.get()`?

Reply:
Don't sweat this too much. You can add a todo note.

If you want to wait until the weekend, I can provide more details, or we can merge this as is

Reply:
We can wait. No rush.

Reply:
@trankchung I can do it potentially. The API underneath the hood is an artifact of when I was new to python (it shows). To answer your question:

-   [TmuxMappingObject.__getattr__](https://github.com/tmux-python/libtmux/blob/v0.15.7/src/libtmux/common.py#L321-L328) (see [python.org](http://python.org/) docs for [Object.__getattr__(self, name)](https://docs.python.org/3/reference/datamodel.html#object.__getattr__))

-   [Window._info](https://github.com/tmux-python/libtmux/blob/v0.15.7/src/libtmux/window.py#L104-L124)

-   [Server._windows](https://github.com/tmux-python/libtmux/blob/v0.15.7/src/libtmux/server.py#L103)* <- *[Server._list_windows()](https://github.com/tmux-python/libtmux/blob/v0.15.7/src/libtmux/server.py#L215-L266)

This API is very old - I will be replacing the code around this to be more transparent
https://github.com/tmux-python/libtmux/pull/426

Reply:
@tony Okay sounds good. Let me know when it's merged and I'll use it for the test.

Reply:
@trankchung How about I merge yours as-is? (so the fix is there?)

Reply:
That #426 PR may not be merged for a while (not sure)

Reply:
> @trankchung How about I merge yours as-is? (so the fix is there?)

As long as you're okay with it.

Reply:
@trankchung I will take a look again in the AM tomorrow 👍 I've tested this by hand so I am feeling optimistic

Reply:
@trankchung I've created #445 to sort out why the test isn't working.  I suspect we're incorrectly filtering it out as a tmux format that was blank, in reality it's an empty string.